tests/queries/0_stateless/02244_url_engine_headers_test.python

```diff
@@ -0,0 +1,141 @@
+import socket
+import sys
+import threading
+import os
+import traceback
+import urllib.request
+import subprocess
+
+
+def is_ipv6(host):
+    try:
+        socket.inet_aton(host)
+        return False
+    except:
+        return True
+
+def get_local_port(host, ipv6):
+    if ipv6:
+        family = socket.AF_INET6
+    else:
+        family = socket.AF_INET
+
+    with socket.socket(family) as fd:
+        fd.bind((host, 0))
+        return fd.getsockname()[1]
+
+CLICKHOUSE_HOST = os.environ.get('CLICKHOUSE_HOST', 'localhost')
+CLICKHOUSE_PORT_HTTP = os.environ.get('CLICKHOUSE_PORT_HTTP', '8123')
+
+# Server returns this JSON response.
+SERVER_JSON_RESPONSE = \
+'''{
+	"login": "ClickHouse",
+	"id": 54801242,
+	"name": "ClickHouse",
+	"company": null
+}'''
+
+EXPECTED_ANSWER = \
+'''{\\n\\t"login": "ClickHouse",\\n\\t"id": 54801242,\\n\\t"name": "ClickHouse",\\n\\t"company": null\\n}'''
+
+#####################################################################################
+# This test starts an HTTP server and serves data to clickhouse url-engine based table.
+# The objective of this test is to check the ClickHouse server provides a User-Agent
+# with HTTP requests.
+# In order for it to work ip+port of http server (given below) should be
+# accessible from clickhouse server.
+#####################################################################################
+
+# IP-address of this host accessible from the outside world. Get the first one
+HTTP_SERVER_HOST = subprocess.check_output(['hostname', '-i']).decode('utf-8').strip().split()[0]
+IS_IPV6 = is_ipv6(HTTP_SERVER_HOST)
+HTTP_SERVER_PORT = get_local_port(HTTP_SERVER_HOST, IS_IPV6)
+
+# IP address and port of the HTTP server started from this script.
+HTTP_SERVER_ADDRESS = (HTTP_SERVER_HOST, HTTP_SERVER_PORT)
+if IS_IPV6:
+    HTTP_SERVER_URL_STR = 'http://' + f'[{str(HTTP_SERVER_ADDRESS[0])}]:{str(HTTP_SERVER_ADDRESS[1])}' + "/"
+else:
+    HTTP_SERVER_URL_STR = 'http://' + f'{str(HTTP_SERVER_ADDRESS[0])}:{str(HTTP_SERVER_ADDRESS[1])}' + "/"
+
+
+def get_ch_answer(query):
+    host = CLICKHOUSE_HOST
+    if IS_IPV6:
+        host = f'[{host}]'
+
+    url = os.environ.get('CLICKHOUSE_URL', 'http://{host}:{port}'.format(host=CLICKHOUSE_HOST, port=CLICKHOUSE_PORT_HTTP))
+    return urllib.request.urlopen(url, data=query.encode()).read().decode()
+
+def check_answers(query, answer):
+    ch_answer = get_ch_answer(query)
+    if ch_answer.strip() != answer.strip():
+        print("FAIL on query:", query, file=sys.stderr)
+        print("Expected answer:", answer, file=sys.stderr)
+        print("Fetched answer :", ch_answer, file=sys.stderr)
+        raise Exception("Fail on query")
+
+class HttpProcessor(SimpleHTTPRequestHandler):
+    def _set_headers(self):
+        test_header = self.headers.get('test_header')
+        if test_header and test_header.startswith('test_header_clickhouse'):
+            self.send_response(200)
+        else:
+            self.send_response(403)
```

Comment:
actual check is here

Reply:
So this test only checks that headers are passed, right? Tests looks too complicated for such simple check, maybe just use the query from den-crane's example? I mean `select * from url(url_mydata, url='http://127.0.0.1:8123?query=select+today(),now()', format='RawBLOB');` or something like that